specification/sbi_atee.adoc

```diff
@@ -1001,32 +860,30 @@ enum Exception {
     STORE_PAGE_FAULT = 15,
     GUEST_INSTRUCTION_PAGE_FAULT = 20,
     /*
-     * The TVM encountered a load fault in a confidential, MMIO, or shared-memory
-     * region. The host should determine the fault address by retrieving the
-     * `htval` from `tvm_vcpu_hypervisor_csrs` and `stval` from `tvm_vcpu_supervisor_csrs`
-     * and combining them as follows: "(htval << 2) | (stval & 0x3)". The fault address
-     * can then be used to determine the type of memory region, and making the appropriate
-     * call (example: sbi_tee_host_add_tvm_zero_pages() to add a demand-zero confidential
-     * page if applicable), and then calling sbi_tee_host_run_tvm_vcpu to resume execution at
-     * the following instruction.
-    */
+     * The TVM encountered a load fault in a confidential, MMIO, or shared memory region. The
+     * host should determine the fault address by retrieving the `htval` and `stval` from NACL
+     * shared memory and combining them as follows: "(htval << 2) | (stval & 0x3)". The fault
+     * address can then be used to determine the type of memory region, and making the appropriate
+     * call (example: sbi_tee_host_add_tvm_zero_pages() to add a demand-zero confidential page if
+     * applicable), and then calling sbi_tee_host_run_tvm_vcpu() to resume execution at the 
+     * following instruction.
+     */
     GUEST_LOAD_PAGE_FAULT = 21,
     /* 
-     * The TVM executed an instruction that caused an exit. The host should decode the
-     * instruction by examining `stval` from `tvm_vcpu_supervisor_csrs`, and determine
-     * the further course of action, and calling then calling sbi_tee_host_run_tvm_vcpu
-     * if appropriate to resume execution at the following instruction.
-    */
+     * The TVM executed an instruction that caused an exit. The host should decode the instruction
+     * by examining `stval` NACL shared memory, and determine the further course of action, and then
+     * calling sbi_tee_host_run_tvm_vcpu() if appropriate to resume execution at the following
+     * instruction.
+     */
     VIRTUAL_INSTRUCTION = 22,
     /*
-     * The TVM encountered a store fault in a confidential, MMIO, or shared-memory
-     * region. The host should determine the fault address by retrieving the
-     * `htval` from `tvm_vcpu_hypervisor_csrs` and `stval` from `tvm_vcpu_supervisor_csrs`
-     * and combining them as follows: "(htval << 2) | (stval & 0x3)". The fault address
-     * can then be used to determine the type of memory region, and making the appropriate
-     * call (example: sbi_tee_host_add_tvm_zero_pages() to add a demand-zero confidential
-     * page if applicable), and then calling sbi_tee_host_run_tvm_vcpu to resume execution at
-     * the following instruction.
+     * The TVM encountered a store fault in a confidential, MMIO, or shared memory region. The
+     * host should determine the fault address by retrieving the `htval` and `stval` from NACL
```

Comment:
Currently, we are mapping the shared memory pages in share/unshare call. It is unlikely that we would get a load/store fault in the shared memory region. In case we get it, host will throw an error and exit to the user space.

Reply:
Fair enough. May be we should explain somewhere the implications of lazy and immediate assignment as an application note.